issue_drafts/issue_4.md

```diff
@@ -0,0 +1,126 @@
+
+For this tutorial, we will choose two lakes that we want to calculate the day-of-year mean temperatures for. The lakes that we want to process have the IDs `120020150` and `107072210`.
+
+Let's process the data based on our current knowledge of Snakemake pipelining. We can write a rule that uses the `2_process/calc_doy_means.py` script. We will need to indicate the input and output files for the two lakes we have chosen in our rule as well:
+```
+rule calc_doy_means:
+    input:
+        "1_fetch/out/tmp/pgdl_nhdhr_120020150_temperatures.csv",
+        "1_fetch/out/tmp/pgdl_nhdhr_107072210_temperatures.csv"
+    output:
+        "2_process/out/doy_120020150.csv",
+        "2_process/out/doy_107072210.csv"
+    script:
+        "2_process/calc_doy_means.py"
+```
+
+This rule is perfectly acceptable and should work if you test it out by running `snakemake --cores 1 calc_doy_means` in your command line/Anaconda prompt.
+
+Note: If you do test out the rule, be sure to delete the outputs before moving forward (`snakemake --cores 1 calc_doy_means --delete-all-output`). If you don't delete the outputs, the Snakemake pipeline will not execute next time because the outputs have already been generated!
+
+Next, let's replace the hardcoded file names and lake IDs in the `2_process/calc_doy_means.py` script with our Snakemake rule properties.
+
+In our Snakefile, we have multiple files as inputs (and multiple files as outputs too), none of which are given variable names in the Snakefile. When we call `snakemake.input` in our Python script, a list of the inputs from the Snakefile will be returned (`["1_fetch/out/tmp/pgdl_nhdhr_120020150_temperatures.csv", "1_fetch/out/tmp/pgdl_nhdhr_107072210_temperatures.csv"]`). Likewise, a list of output file names will be returned when we call `snakemake.output`.
+
+Since the Snakemake rule inputs and outputs already have the lake IDs included, we can remove the list of `lake_ids` from our script. However, since we were using the lake IDs to indicate how many iterations our for loop needs to go through, we will now need to change that. We can choose the number of iterations by using the length of our input file list: `for i in range(len(snakemake.input))`. We can call each input and output file name from the list by index as we go through our for loop.
+
+Let's put that all together:
+```
+if __name__ == '__main__':
+    for i in range(len(snakemake.input))
+        out_file = snakemake.output[i]
+        in_file = snakemake.input[i]
+```
+
+Lastly, let's extract the `lake_id` from our `out_file` name in each iteration of the for loop. Here's the full code block to update your `calc_doy_means.py` script with:
+```
+if __name__ == '__main__':
+    for i in range(len(snakemake.input)):
+        out_file = snakemake.output[i]
+        in_file = snakemake.input[i]
+        lake_id = os.path.splitext(out_file)[0].split('_')[-1]
+        main(out_file, in_file, lake_id)
+```
```

Comment:
I like the scaffolded learning here - I think looping over multiple inputs is a nice intermediate step!